src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FallThroughCheck.java

```diff
@@ -486,46 +496,74 @@ private boolean checkSynchronized(final DetailAST synchronizedAst, boolean useBr
         else {
-            final int startLineNo = currentCase.getLineNo();
-            for (int i = endLineNo - 2; i > startLineNo - 1; i--) {
-                final int[] line = getLineCodePoints(i);
-                if (!CodePointUtil.isBlank(line)) {
-                    allThroughComment = matchesComment(reliefPattern, i + 1);
-                    break;
-                }
+            ast = currentCase;
+            lastAST = currentCase.getParent().getLastChild();
+        }
+        boolean reliefComment = false;
+        final List<DetailAST> commentContent = getCommentContentNodes(ast, lastAST);
+        for (DetailAST detailAST : commentContent) {
+            if (reliefPattern.matcher(detailAST.getText()).find()) {
+                reliefComment = true;
             }
         }
-        return allThroughComment;
+        return reliefComment || commentAsChildOfSameGroup(reliefPattern, currentCase);
     }
 
     /**
-     * Does a regular expression match on the given line and checks that a
-     * possible match is within a comment.
+     * This method will check if the comment are child of same case group ast.
      *
-     * @param pattern The regular expression pattern to use.
-     * @param lineNo The line number in the file.
-     * @return True if a match was found inside a comment.
+     * @param reliefPattern The regular expression pattern to match the comment text
+     * @param currentCase AST of the case that falls through to the next case.
+     * @return true if relief comment found
      */
-    // suppress deprecation until https://github.com/checkstyle/checkstyle/issues/11166
-    @SuppressWarnings("deprecation")
-    private boolean matchesComment(Pattern pattern, int lineNo) {
-        final String line = getLine(lineNo - 1);
-
-        final Matcher matcher = pattern.matcher(line);
-        return matcher.find()
-                && getFileContents().hasIntersectionWithComment(
-                        lineNo, matcher.start(), lineNo, matcher.end());
+    private static boolean commentAsChildOfSameGroup(Pattern reliefPattern,
+                                                     DetailAST currentCase) {
+        boolean reliefComment = false;
+        final List<DetailAST> commentContent = getCommentContentNodes(currentCase.getLastChild(),
+                currentCase.getLastChild().getLastChild());
+        for (DetailAST detailAST : commentContent) {
+            if (reliefPattern.matcher(detailAST.getText()).find()) {
+                reliefComment = true;
+            }
+        }
+        return reliefComment;
     }
 
+    /**
+     * This method will return the comment content node between two nodes.
+     *
+     * @param fromNode starting node
+     * @param toNode last node to check
+     * @return list of commentContent node
+     */
+    private static List<DetailAST> getCommentContentNodes(DetailAST fromNode, DetailAST toNode) {
+        final List<DetailAST> commentContentNode = new LinkedList<>();
```

Comment:
`final List<DetailAST> commentContentNode`

collection should be plural noun `...Notes`.
but name is too long, name it `result` to not repeat name of method.

Reply:
> collection should be plural noun `...Notes`. but name is too long, name it `result` to not repeat name of method.

will take care of this minor mistake 

name changed as result 

done